components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -218,6 +220,13 @@ class Session(
         notifyObservers(old, new) { onNavigationStateChanged(this@Session, canGoBack, new) }
     }
 
+    /**
+     * Local session history, with the current page selected.
+     */
+    var historyList: List<HistoryItem> by Delegates.observable(emptyList()) { _, old, new ->
+        notifyObservers(old, new) { onHistoryStateChanged(this@Session, new) }
+    }
```

Comment:
naming: nit: I would suggest to only name this `history`. In addition to that I wonder if a rich state object (a bit like GeckoView's HistoryList) would be nicer to use: We can later attach things like canGoForward/canGoBack all to this `SessionHistory` object. That's at least how I would do it for `browser-state`.